probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -123,7 +123,7 @@ pub struct ArmCommunicationInterface {
 }
 
 impl ArmCommunicationInterface {
-    pub fn new(probe: Probe) -> Result<Self, DebugProbeError> {
+    pub fn new(probe: Probe) -> Result<Self, (Probe, DebugProbeError)> {
```

Comment:
Why was this change made? I don't see why this is necessary somehow ...

Reply:
If creating the interface fails, you might still want to use the `Probe`. But if we don't return it, you cannot access it anymore.

I was running into this problem with the auto-detection, if something fails you might want to try another interface, which was not possible so far.

Reply:
Ok, sounds goot to me :)